packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -235,11 +235,25 @@ export class FileStorage implements DStore {
 
     // if have children, keep this node around as a stub
     if (!_.isEmpty(noteToDelete.children)) {
-      this.logger.info({ ctx, noteAsLog, msg: "keep as stub" });
-      const prevNote = { ...noteToDelete };
-      noteToDelete.stub = true;
-      this.updateNote(noteToDelete);
-      out.push({ note: noteToDelete, status: "update", prevNote });
+      if (!opts?.replaceWithNewStub) {
```

Comment:
the issue is that if the note being deleted has children, we still might not want to replace it. it is valid to rename a note to itself for the "rename header" operation. this is why `deleteNote` needs to know about this

this should be a private function - we should have a public `/note/delete` method that calls something higher level than this function

Reply:
In the same vein, is there any risk to always doing this? Is updating the old note with `stub: true` a performance optimization?

Reply:
Now that I look at it, I _think_ there's no reason to not always do this like @SeriousBug says. The change status being `update` is also a little weird too given that we are deleting it.

Reply:
@kevinslin actually, `deleteNote` doesn't need to take into account of the rename header operation case: in `renameNote`, when the old note and new note has the same fname, we don't call `deleteNote` at all.

Reply:
>  Is updating the old note with stub: true a performance optimization?

Looks like not doing it like this would cause regular deletes (not coming from rename) not work.
Could probably fix that to work properly in some way, but not in the scope of this PR, so I'll defer looking into that

Reply:
I see - in that case, can we at least put a @deprecated tag or a TODO on the `replaceWithNewStub` property? 

Sorry for being nitpicky here - but I want to get the engine API into a more precise state - our API functions and their parameters are too loose right now, and I think this is part of the reason why we have more engine state issues now - it's not clear to callers which flags are supposed to be set and why, and in some cases the API is exposing things that clients shouldn't need to care about.

Reply:
@jonathanyeung Agreed.

Discussed with @kevinslin as well about the current state of engine API code being a bit too stateful, and we would ideally want these to be clearly specced out and cleaned up in the future. I'll add a comment in the docstring